server/util.go

```diff
@@ -245,75 +245,72 @@ func uniformValue(value interface{}) interface{} {
 	}
 }
 
-func dumpRowValuesBinary(alloc arena.Allocator, columns []*ColumnInfo, row []types.Datum) (data []byte, err error) {
+func dumpRowValuesBinary(alloc arena.Allocator, columns []*ColumnInfo, row []types.Datum, buffer []byte) ([]byte, error) {
```

Comment:
I think put `buffer` to the first argument is better.
`alloc` can be removed if all the `dump` methods use `buffer` to append the value.